test/unit/mdc-tabs/mdc-tab.test.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assert} from 'chai';
+import bel from 'bel';
+import domEvents from 'dom-events';
+import td from 'testdouble';
+import {createMockRaf} from '../helpers/raf';
+import {supportsCssVariables} from '../../../packages/mdc-ripple/util';
+import {MDCTab} from '../../../packages/mdc-tabs/tab';
+import {MDCTabFoundation} from '../../../packages/mdc-tabs/tab';
+import {cssClasses} from '../../../packages/mdc-tabs/tab/constants';
+
+function getFixture() {
+  return bel`
+    <div>
+      <a class="mdc-tab mdc-tab--active" href="#one">Item One</a>
+    </div>`;
+}
+
+function setupTest() {
+  const fixture = getFixture();
+  const root = fixture.querySelector('.mdc-tab');
+  const component = new MDCTab(root);
+  return {fixture, root, component};
+}
+
+suite('MDCTab');
+
+test('attachTo returns a component instance', () => {
+  assert.isOk(MDCTab.attachTo(getFixture()) instanceof MDCTab);
+});
+
+if (supportsCssVariables(window)) {
+  test('#destroy cleans up ripple on tab', () => {
+    const raf = createMockRaf();
+    const {root, component} = setupTest();
+
+    raf.flush();
+
+    component.destroy();
+    raf.flush();
+
+    assert.isFalse(root.classList.contains('mdc-ripple-upgraded'));
+    raf.restore();
+  });
+}
+
+test('#get computedWidth returns computed width of tab', () => {
+  const {root, component} = setupTest();
+
+  assert.equal(component.computedWidth, root.offsetWidth);
+});
+
+test('#get computedLeft returns computed left offset of tab', () => {
+  const {root, component} = setupTest();
+
+  assert.equal(component.computedLeft, root.offsetLeft);
+});
+
+test('#get/set isActive', () => {
+  const {component} = setupTest();
+
+  component.isActive = false;
+  assert.isFalse(component.root_.classList.contains(cssClasses.ACTIVE));
+
+  component.isActive = true;
+  assert.isTrue(component.root_.classList.contains(cssClasses.ACTIVE));
+});
+
+test('#get/set preventDefaultOnClick', () => {
```

Comment:
this should also test that the foundation logic from `preventDefaultOnClick` kicks in